clang/test/dpct/helper_files_ref/include/memory.hpp

```diff
@@ -1245,6 +1245,51 @@ template <class T, size_t Dimension>
 using constant_memory = detail::device_memory<T, constant, Dimension>;
 template <class T, size_t Dimension>
 using shared_memory = detail::device_memory<T, shared, Dimension>;
-} // namespace dpct
 
+class pointer_attributes {
+public:
+  void init(const void *ptr,
+              sycl::queue &q = dpct::get_default_queue()) {
+#ifdef DPCT_USM_LEVEL_NONE
+    dpct::detail::pointer_access_attribute attribute =
+             dpct::detail::get_pointer_attribute(q, ptr);
+    if (attribute == dpct::detail::pointer_access_attribute::host_only) {
+      memory_type = sycl::usm::alloc::unknown;
+      device_pointer = nullptr;
+      host_pointer = nullptr;
+    } else {
+      if (attribute ==
+          dpct::detail::pointer_access_attribute::device_only) {
+        memory_type = sycl::usm::alloc::device;
+        host_pointer = nullptr;
+      } else {
+        host_pointer = nullptr;
+        memory_type = sycl::usm::alloc::shared;
+        is_shared = 0;
+      }
+      auto accessor = dpct::detail::mem_mgr::instance().
+                        translate_ptr(ptr).buffer.get_access();
```

Comment:
get_access() is deprecated,  don't use it. ref to spec in P109.

In USM none mode,  buffer is backed by host,  while it's device ptr is runtime allocated.  Suggest not to support for USM node mode.